pkg/logs/input/file/tailer_key.go

```diff
@@ -0,0 +1,19 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package file
+
+import "fmt"
+
+// tailerKey can be implemented by File and Tailer
+type tailerKey interface {
```

Comment:
About this interface, why not only having a `Key() string` method? Do you have other plans for `getPath()` and `getSourceIdentifier()`?

That would not avoid the extra `buildTailerKey()` function but simplify the role of the interface.

Reply:
yes I considered that option and preferred to keep the key generating logic isolated and in one place not duplicated in the file_provider and the tailer.
I thought it's better this way because we want to make sure file_provider and tailer always return the same key, in other words, it safer to let the scanner build its own map keys and IMHO more robust to future changes in file_provider and tailer.
wdyt?

Reply:
Sounds totally sensible! 👍